hybrid/src/ecies_aead_hkdf_private_key_manager.rs

```diff
@@ -0,0 +1,230 @@
+            return Err("EciesAeadHkdfPrivateKeyKeyManager: invalid key format".into());
+        }
+        let key_format = tink_proto::EciesAeadHkdfKeyFormat::decode(serialized_key_format)
+            .map_err(|e| wrap_err("EciesAeadHkdfPrivateKeyKeyManager: invalid key format", e))?;
+        let (_pt_format, curve, _hash, _kem_params, _dem_params, _aead_dem) =
+            validate_key_format(&key_format)
+                .map_err(|e| wrap_err("EciesAeadHkdfPrivateKeyKeyManager", e))?;
+        let pvt = crate::subtle::generate_ecdh_key_pair(curve)?;
+        let (x, y) = pvt
+            .public_key()
+            .x_y_bytes()
+            .map_err(|e| wrap_err("EciesAeadHkdfPrivateKeyKeyManager", e))?;
+
+        let priv_key = tink_proto::EciesAeadHkdfPrivateKey {
+            version: ECIES_AEAD_HKDF_PRIVATE_KEY_KEY_VERSION,
+            key_value: pvt.d_bytes(),
+            public_key: Some(tink_proto::EciesAeadHkdfPublicKey {
+                version: ECIES_AEAD_HKDF_PRIVATE_KEY_KEY_VERSION,
+                params: key_format.params,
+                x,
+                y,
+            }),
+        };
+        let mut sk = Vec::new();
+        priv_key.encode(&mut sk).map_err(|e| {
+            wrap_err(
+                "EciesAeadHkdfPrivateKeyKeyManager: failed to encode new key",
+                e,
+            )
+        })?;
+        Ok(sk)
+    }
+
+    fn type_url(&self) -> &'static str {
+        ECIES_AEAD_HKDF_PRIVATE_KEY_TYPE_URL
+    }
+
+    fn key_material_type(&self) -> tink_proto::key_data::KeyMaterialType {
+        tink_proto::key_data::KeyMaterialType::AsymmetricPrivate
+    }
+
+    fn supports_private_keys(&self) -> bool {
+        true
+    }
+
+    fn public_key_data(
+        &self,
+        serialized_priv_key: &[u8],
+    ) -> Result<tink_proto::KeyData, TinkError> {
+        let priv_key = tink_proto::EciesAeadHkdfPrivateKey::decode(serialized_priv_key)
+            .map_err(|e| wrap_err("EciesAeadHkdfPrivateKeyKeyManager: invalid private key", e))?;
+        let mut serialized_pub_key = Vec::new();
+        priv_key
+            .public_key
+            .ok_or_else(|| TinkError::new("EciesAeadHkdfPrivateKeyKeyManager: no public key"))?
+            .encode(&mut serialized_pub_key)
+            .map_err(|e| wrap_err("EciesAeadHkdfPrivateKeyKeyManager: invalid public key", e))?;
+        Ok(tink_proto::KeyData {
+            type_url: crate::ECIES_AEAD_HKDF_PUBLIC_KEY_TYPE_URL.to_string(),
+            value: serialized_pub_key,
+            key_material_type: tink_proto::key_data::KeyMaterialType::AsymmetricPublic as i32,
+        })
+    }
+}
+
+/// Validate the given [`tink_proto::EciesAeadHkdfPrivateKey`] and return the parameters.
+fn validate_key(
+    key: &tink_proto::EciesAeadHkdfPrivateKey,
+) -> Result<
+    (
+        EcPointFormat,
+        EllipticCurveType,
+        HashType,
+        &EciesHkdfKemParams,
+        &EciesAeadDemParams,
+        &KeyTemplate,
```

Comment:
Maybe, but a struct would end up being just a copy of `EciesAeadHkdfParams` but with the `Option`s removed and the `i32`s converted to the relevant `enum`s &ndash; which seems like it might confuse more than help.

Given that this is internal, I'm going to leave as-is, but I'll drop the `EciesAeadDemParams` return value (as  it's just a container for the `KeyTemplate`).